kaldb/src/test/java/com/slack/kaldb/logstore/FieldConflictsTest.java

```diff
@@ -0,0 +1,227 @@
+package com.slack.kaldb.logstore;
+
+import static com.slack.kaldb.testlib.TemporaryLogStoreAndSearcherRule.findAllMessages;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import brave.Tracing;
+import com.slack.kaldb.testlib.MessageUtil;
+import com.slack.kaldb.testlib.TemporaryLogStoreAndSearcherRule;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Map;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class FieldConflictsTest {
+  @BeforeClass
+  public static void beforeClass() throws Exception {
+    Tracing.newBuilder().build();
+  }
+
+  @Rule
+  public TemporaryLogStoreAndSearcherRule strictLogStore =
+      new TemporaryLogStoreAndSearcherRule(false);
+
+  public FieldConflictsTest() throws IOException {}
+
+  @Test
+  public void testFieldConflictingFieldTypeWithSameValue() throws InterruptedException {
+    final String conflictingFieldName = "conflictingField";
+
+    LogMessage msg1 =
+        new LogMessage(
+            MessageUtil.TEST_INDEX_NAME,
+            "INFO",
+            "1",
+            Map.of(
+                LogMessage.ReservedField.TIMESTAMP.fieldName,
+                MessageUtil.getCurrentLogDate(),
+                LogMessage.ReservedField.MESSAGE.fieldName,
+                "Test message",
+                LogMessage.ReservedField.TAG.fieldName,
+                "foo-bar",
+                LogMessage.ReservedField.HOSTNAME.fieldName,
+                "host1-dc2.abc.com",
+                conflictingFieldName,
+                "1"));
+    strictLogStore.logStore.addMessage(msg1);
+
+    LogMessage msg2 =
+        new LogMessage(
+            MessageUtil.TEST_INDEX_NAME,
+            "INFO",
+            "2",
+            Map.of(
+                LogMessage.ReservedField.TIMESTAMP.fieldName,
+                MessageUtil.getCurrentLogDate(),
+                LogMessage.ReservedField.MESSAGE.fieldName,
+                "Test message",
+                LogMessage.ReservedField.TAG.fieldName,
+                "foo-bar",
+                LogMessage.ReservedField.HOSTNAME.fieldName,
+                "host1-dc2.abc.com",
+                conflictingFieldName,
+                1));
+    strictLogStore.logStore.addMessage(msg2);
+
+    strictLogStore.logStore.commit();
+    strictLogStore.logStore.refresh();
+    Thread.sleep(1000);
```

Comment:
This should be converted to an `await()` - we have metrics on the log store that should be emitted once the commit and refresh are complete.

Reply:
Done.